src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -34,13 +35,16 @@ public static int Main(string[] args)
 
             app.OnExecute(() =>
             {
-                if (!CheckArg(framework))
+                if (framework.HasValue() && IsUnsupportedFramework(framework.Value()))
```

Comment:
yea, I agree. Some code has to deal with these nullable values and I'd prefer to keep it as close to the Command Line Parser as possible.